src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -579,19 +584,35 @@ public void TotalProcessorTime_GetNotStarted_ThrowsInvalidOperationException()
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/corefx/issues/22266", TargetFrameworkMonikers.UapNotUapAot)]
         public void TestProcessStartTime()
         {
-            TimeSpan allowedWindow = TimeSpan.FromSeconds(3);
-            DateTime testStartTime = DateTime.UtcNow;
-            using (var remote = RemoteInvoke(() => { Console.Write(Process.GetCurrentProcess().StartTime.ToUniversalTime()); return SuccessExitCode; },
-                new RemoteInvokeOptions { StartInfo = new ProcessStartInfo { RedirectStandardOutput = true } }))
+            TimeSpan allowedWindow = TimeSpan.FromSeconds(1);
+
+            for (int i = 0; i < 2; i++)
             {
-                Assert.Equal(remote.Process.StartTime, remote.Process.StartTime);
+                Process p = CreateProcessPortable(RemotelyInvokable.ReadLine);
 
-                DateTime remoteStartTime = DateTime.Parse(remote.Process.StandardOutput.ReadToEnd());
-                DateTime curTime = DateTime.UtcNow;
-                Assert.InRange(remoteStartTime, testStartTime - allowedWindow, curTime + allowedWindow);
+                Assert.Throws<InvalidOperationException>(() => p.StartTime);
+
+                DateTime testStartTime = DateTime.Now;
+                p.StartInfo.RedirectStandardInput = true;
+                p.Start();
+                Assert.Equal(p.StartTime, p.StartTime);
+                DateTime processStartTime = p.StartTime;
+                using (StreamWriter writer = p.StandardInput)
+                {
+                    writer.WriteLine("start");
+                }
+
+                Assert.True(p.WaitForExit(WaitInMS));
+                DateTime testEndTime = DateTime.Now;
+
+                bool hasTimeChanged = testEndTime < testStartTime;
+                if (i != 0 || !hasTimeChanged)
```

Comment:
It might be a little more readable if you simply did
```c#
if (!hasTimeChanged)
    break;

} // end loop

Assert.InRange(processStartTime, testStartTime - allowedWindow, testEndTime + allowedWindow);
```